services/supervisor/src/lib.rs

```diff
@@ -69,34 +98,34 @@ fn update_configs(context: &mut CallContext<'_>, changes: Vec<ConfigChange>) {
     changes.into_iter().for_each(|change| match change {
         ConfigChange::Consensus(config) => {
             log::trace!("Updating consensus configuration {:?}", config);
+            let configure_result = apply_consensus_config(context, config);
 
-            let result = context.isolate(|context| {
-                blockchain::Schema::new(context.fork())
-                    .consensus_config_entry()
-                    .set(config);
-                Ok(())
-            });
-            assert!(result.is_ok());
+            if let Err(e) = configure_result {
+                // Consensus configuration is not allowed to fail, since it was validated and approved.
+                // This is an unrecoverable error.
+                panic!("Consensus configuration failed with error {:?}", e)
+            }
         }
 
         ConfigChange::Service(config) => {
             log::trace!(
                 "Updating service instance configuration, instance ID is {}",
                 config.instance_id
             );
+            let configure_result = apply_service_config(context, config);
```

Comment:
Why not shorter form?
```rust
if let Err(e) = apply_service_config(context, config) { ... }
```

Reply:
To have the invocation more visible. Don't think that here it improves the readability.